rework navigation component to reposition the name-box and save status

MysteriousChallenger opened the pull request:
![image](https://user-images.githubusercontent.com/55671762/101298103-3a498c80-37e1-11eb-85c3-a6859627fb47.png)
comments welcome

Yicong-Huang commented:
The design looks way better than before! Thanks. However, there are some issues that I identified when testing:
1. When the focus goes onto the workflow name input box, it shrinks the size to be too small:
<img width="295" alt="Screen Shot 2020-12-06 at 5 49 13 PM" src="https://user-images.githubusercontent.com/17627829/101300730-623ded80-37eb-11eb-92c3-8bf2c3eadc6a.png">
2. When typing in the input box, you can see the cursor is out of the input box:
<img width="328" alt="Screen Shot 2020-12-06 at 5 47 23 PM" src="https://user-images.githubusercontent.com/17627829/101300742-6ec24600-37eb-11eb-9030-8ef8e83ed0a0.png">
3. When resizing the window, the control buttons and the operator menu are starting to disappear. I don't like this, @zuozhiw @avinash0161 what do you think?
<img width="572" alt="Screen Shot 2020-12-06 at 5 51 24 PM" src="https://user-images.githubusercontent.com/17627829/101300802-afba5a80-37eb-11eb-830d-32432ba49120.png"> (See that the `Run` button and operators penal are disappearing.

Yicong-Huang commented on core/new-gui/src/app/workspace/component/navigation/navigation.component.ts:
```diff
@@ -295,8 +309,8 @@ export class NavigationComponent implements OnInit {
-  onWorkflowNameChange() {
-    this.currentWorkflowName = this.currentWorkflowName.trim();
+  onWorkflowNameChange(event: InputEvent) {
+    this.currentWorkflowName = (<HTMLElement>event.target).innerText.trim();
```
why do we need to use native HTML elements to do this?

zuozhiw replied:
See the two-way data binding comment in the corresponding html element

MysteriousChallenger replied:
I think this is a bug with Angular:
two-way binding to content just doesn't work with the contenteditable="true" attribute. Editing content doesn't trigger the event listener to update this.currentWorkflowName. Manually triggering the event listener still works though.

To fix this, I'd need to use an <input> tag. However <input> tags don't expand when you type into them, and I can't replicate that effect without some fancy JS

Yicong-Huang reviewed (changes requested):
Left some comments, please take care of them

Yicong-Huang commented:
@MysteriousChallenger #924 changed version number component already. Let's sync with that.

zuozhiw commented on core/new-gui/src/app/workspace/component/navigation/navigation.component.html:
```diff
@@ -1,13 +1,26 @@
+          <span class="title">Texera</span>
+          <span class="version-number">0.5.1</span>
       </a>
+      <div class="git-commit-hash-label">
```
this is already solved by #924

zuozhiw commented on core/new-gui/src/app/workspace/component/navigation/navigation.component.html:
```diff
@@ -1,13 +1,26 @@
     </div>
+
+    <div class="texera-navigation-workflow-name" *ngIf="userSystemEnabled">
+      <span class="name-input" #nameInput role="textbox" contenteditable="true" spellcheck="false" placeholder="Untitled Workflow" (input)="onWorkflowNameChange($event)" ></span>
```
I think it's good to just use the old two-way data binding. Is there something wrong with it?
`<input (change)="onWorkflowNameChange()" [(ngModel)]="currentWorkflowName"`

Yicong-Huang replied:
agreed, @MysteriousChallenger ?

MysteriousChallenger replied:
https://github.com/Texera/texera/pull/923#discussion_r540591787

zuozhiw reviewed (approved):
Looks good, left a minor comment

MysteriousChallenger commented:
In reply to review https://github.com/Texera/texera/pull/923#issuecomment-739616366
- added a minimum width to the name input
- save status smaller and grey (the save message itself is pretty long, but that can be changed later)
- removed extra version number in the navigator

Yicong-Huang reviewed (approved):
I think it's good. @MysteriousChallenger please merge.
